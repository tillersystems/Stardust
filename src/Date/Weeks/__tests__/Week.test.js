import React from 'react';
import { DateTime } from 'luxon';
import { fireEvent } from 'react-testing-library';

import Week from '..';

describe('<Weeks />', () => {
  const dateValue = DateTime.fromObject({
    year: 2018,
    month: 7,
    day: 15,
    hour: 0,
    minute: 0,
    second: 0,
    zone: 'Europe/Paris',
  });

  test('should render without a problem', () => {
    const { container } = render(
      <Week
        currentMonth={dateValue}
        onDateClick={() => {}}
        selectedDate={dateValue}
        minDate={dateValue}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should handle onDateClick', () => {
    const dateSelected = DateTime.fromObject({
      year: 2018,
      month: 7,
      day: 20,
      hour: 0,
      minute: 0,
      second: 0,
      zone: 'Europe/Paris',
    });
    const spy = jest.fn();
    const { getByText } = render(
      <Week
        currentMonth={dateValue}
        onDateClick={spy}
        selectedDate={dateValue}
        minDate={dateValue}
      />,
    );

    const dayNode = getByText('20');

    // Click on a day
    fireEvent.click(dayNode);

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(dateSelected);
  });

  test('should not handle onDateClick if the selected day is before min date', () => {
    const spy = jest.fn();
    const { getByText } = render(
      <Week
        currentMonth={dateValue}
        onDateClick={spy}
        selectedDate={dateValue}
        minDate={dateValue}
      />,
    );

    const dayNode = getByText('10');

    // Click on a day
    fireEvent.click(dayNode);

    expect(spy).toHaveBeenCalledTimes(0);
  });

  test('should not handle onDateClick if the selected day is after max date', () => {
    const spy = jest.fn();
    const { getByText } = render(
      <Week
        currentMonth={dateValue}
        onDateClick={spy}
        selectedDate={dateValue}
        maxDate={dateValue}
      />,
    );

    const dayNode = getByText('20');

    // Click on a day
    fireEvent.click(dayNode);

    expect(spy).toHaveBeenCalledTimes(0);
  });
});
