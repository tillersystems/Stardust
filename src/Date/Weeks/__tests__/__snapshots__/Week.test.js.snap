// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weeks /> should render withouth a problem 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c1 > .c2 {
  color: hsl(207,22%,90%);
}

.c1 > .c2 {
  color: hsl(207,22%,90%);
}

.c4 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
}

.c4 > .c2 {
  color: hsl(207,22%,90%);
}

.c4 > .c2 {
  color: hsl(207,22%,90%);
}

.c6 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c6 > .c2 {
  color: hsl(207,22%,90%);
}

.c8 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c8 > .c2 {
  color: hsl(207,22%,90%);
}

.c9 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
}

.c9 > .c2 {
  color: hsl(207,22%,90%);
}

.c10 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c10 > .c2 {
  color: hsl(207,22%,90%);
}

.c12 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c13 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
}

.c14 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c15 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
}

.c15 > .c2 {
  color: hsl(207,22%,90%);
}

.c16 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c16 > .c2 {
  color: hsl(207,22%,90%);
}

.c3 {
  width: 100%;
  height: 100%;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c5 {
  width: 100%;
  height: 100%;
}

.c7 {
  width: 100%;
  height: 100%;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c11 {
  width: 100%;
  height: 100%;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(7,2.8rem);
  grid-template-rows: 2.8rem;
  justify-items: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  text-align: center;
  color: hsl(206,23%,69%);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      disabled=""
    >
      <div
        class="c2 c3"
      >
        25
      </div>
    </div>
    <div
      class="c4"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        26
      </div>
    </div>
    <div
      class="c4"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        27
      </div>
    </div>
    <div
      class="c4"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        28
      </div>
    </div>
    <div
      class="c4"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        29
      </div>
    </div>
    <div
      class="c4"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        30
      </div>
    </div>
    <div
      class="c6"
      disabled=""
    >
      <div
        class="c2 c7"
      >
        01
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c8"
      disabled=""
    >
      <div
        class="c2 c3"
      >
        02
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        03
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        04
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        05
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        06
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        07
      </div>
    </div>
    <div
      class="c10"
      disabled=""
    >
      <div
        class="c2 c11"
      >
        08
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c8"
      disabled=""
    >
      <div
        class="c2 c3"
      >
        09
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        10
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        11
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        12
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        13
      </div>
    </div>
    <div
      class="c9"
      disabled=""
    >
      <div
        class="c2 c5"
      >
        14
      </div>
    </div>
    <div
      class="c10"
      disabled=""
    >
      <div
        class="c2 c11"
      >
        15
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c12"
    >
      <div
        class="c2 c3"
      >
        16
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        17
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        18
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        19
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        20
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        21
      </div>
    </div>
    <div
      class="c14"
    >
      <div
        class="c2 c11"
      >
        22
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c12"
    >
      <div
        class="c2 c3"
      >
        23
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        24
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        25
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        26
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        27
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c2 c5"
      >
        28
      </div>
    </div>
    <div
      class="c14"
    >
      <div
        class="c2 c11"
      >
        29
      </div>
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c12"
    >
      <div
        class="c2 c3"
      >
        30
      </div>
    </div>
    <div
      class="c14"
    >
      <div
        class="c2 c11"
      >
        31
      </div>
    </div>
    <div
      class="c15"
    >
      <div
        class="c2 c5"
      >
        01
      </div>
    </div>
    <div
      class="c15"
    >
      <div
        class="c2 c5"
      >
        02
      </div>
    </div>
    <div
      class="c15"
    >
      <div
        class="c2 c5"
      >
        03
      </div>
    </div>
    <div
      class="c15"
    >
      <div
        class="c2 c5"
      >
        04
      </div>
    </div>
    <div
      class="c16"
    >
      <div
        class="c2 c11"
      >
        05
      </div>
    </div>
  </div>
</div>
`;
